server/py/framework/db/sqldb/db.py

```diff
@@ -3863,6 +3891,48 @@ def _calculate_feature_sets_counters(session) -> dict[str, int]:
         }
         return project_to_feature_set_count
 
+    def _calculate_mm_functions_counters(
+        self, session
+    ) -> tuple[dict[str, int], dict[str, int]]:
+        labels = [f"{ModelMonitoringAppLabel.KEY}={ModelMonitoringAppLabel.VAL}"]
+        query = session.query(Function.project, Function, Function.Tag.name)
+        query = query.join(
+            Function.Tag, Function.id == Function.Tag.obj_id
+        )  # filter duplications
+        labels = label_set(labels)
+        query = self._add_labels_filter(session, query, Function, labels)
+
+        project_to_failed_mm_functions_count = {}
+        project_to_running_mm_functions_count = {}
+        for project, function, name in query.all():
+            project_to_running_mm_functions_count.setdefault(project, 0)
+            project_to_failed_mm_functions_count.setdefault(project, 0)
+            if function.state == mlrun.common.schemas.FunctionState.ready:
+                project_to_running_mm_functions_count[project] += 1
+            if function.state == mlrun.common.schemas.FunctionState.error:
+                project_to_failed_mm_functions_count[project] += 1
+
```

Comment:
this could be done on sql side using count on columns and filters on states
the reason to do it is to keep mlrun api memory low to avoid bringing all project mm functions to python side for calculations

* we do need to add index on function projectXstate for such usecase